### DISCUSSION TITLE
Is there a better way to handle errors thrown by Prisma?

### DISCUSSION THREAD

**Discussion by harshPPatel** [trpc/trpc#3715] [Q&A] [answered]
So currently, In my TRPC functions, i have code like this all over the place (sometimes, multiple times in one function/endpoint).
```ts
import { TRPCError } from "@trpc/server";
import { protectedProcedure, router } from "../trpc";

export const userRouter = router({
  me: protectedProcedure.query(async ({ ctx }) => {
    const dbuser = await ctx.prisma.user.findUnique({ where: { id: "ctx.session.user.id" } });
    if (!dbuser) {
      throw new TRPCError({
        code: "NOT_FOUND",
      });
    }
    return dbuser;
  }),
});
```

I like how the code is explicit so that I can see what and where I am throwing the `NOT_FOUND` error but soon it gets annoying and makes my functions quite long. 

I was thinking to refactor such code to something like this:
```ts
import { protectedProcedure, router } from "../trpc";

export const userRouter = router({
  me: protectedProcedure.query(async ({ ctx }) => {
    // Primsa throws error if record is not found. Currently, in Create-T3-App, it returns proper error message, but the status code is "INTERNAL_SERVER_ERROR"
    return ctx.prisma.user.findUniqueOrThrow({ where: { id: "ctx.session.user.id" } }); 
  }),
});
```

This is where I am getting confused about what is a good practice according to the TRPC team and community. Should I keep using the first pattern where code is explicit on where an error is being thrown? Or should I be using the second approach with some hack where I can add some magic in the `onError` hook to return proper error codes?

I would really appreciate any help on this question. Thank you!

**Comment by KATT** [comment, marked as answer]
You can do ` ?? throwNotFound()`

```ts
import { TRPCError } from '@trpc/server';

export function throwNotFound(message?: string) {
  throw TRPCError({ code: 'NOT_FOUND', message: message ?? 'Not found' })
}
```

**Reply by harshPPatel** [reply]
That can work but I meant to replace this logic:
```ts
const dbuser = await ctx.prisma.user.findUnique({ where: { id: "ctx.session.user.id" } }); // <-- I run Prisma Query
    if (!dbuser) { // <-- I have to manually check and throw error
      throw new TRPCError({
        code: "NOT_FOUND",
      });
    }
```

If the TRPC community counts this as a good practice, I am happy to comply but I just wanted to know if there were better options 😄

**Reply by KATT** [reply]
```ts
const dbuser = (await ctx.prisma.user.findUnique({ where: { id: "ctx.session.user.id" } })) ?? throwNotFound()

```
